vital/bindings/python/vital/tests/test_config.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python
+#ckwg +28
+# Copyright 2019 by Kitware, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#  * Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+#  * Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+#  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+#    to endorse or promote products derived from this software without specific
+#    prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+from vital.config import config
+import nose.tools
+
+class TestVitalConfig(object):
+    def test_create(self):
+        nose.tools.ok_(len(config.empty_config()) == 0)
+        config.ConfigKeys()
+
+    def test_api_calls(self):
+        config.Config.block_sep
+        config.Config.global_value
+
+    def test_has_value(self):
+        c = config.empty_config()
+        keya = 'keya'
+        keyb = 'keyb'
+        valuea = 'valuea'
+        c.set_value(keya, valuea)
+        nose.tools.ok_(c.has_value(keya))
+        nose.tools.ok_(not c.has_value(keyb))
+
+    def test_get_value(self):
+        c = config.empty_config()
+        keya = 'keya'
+        valuea = 'valuea'
+        c.set_value(keya, valuea)
+        get_valuea = c.get_value(keya)
+        nose.tools.assert_equal(valuea, get_valuea)
+
+    def test_get_value_nested(self):
+        c = config.empty_config()
+        keya = 'keya'
+        keyb = 'keyb'
+        valuea = 'valuea'
+        c.set_value(keya + config.Config.block_sep + keyb, valuea)
+        nc = c.subblock(keya)
+        get_valuea = nc.get_value(keyb)
+        nose.tools.assert_equal(valuea, get_valuea)
+
+    @nose.tools.raises(BaseException)
+    def test_get_value_no_exist(self):
+        c = config.empty_config()
+        keya = 'keya'
+        keyb = 'keyb'
+        valueb = 'valueb'
+        c.get_value(keya)
```

Comment:
Minor, but it would be good to remove any unused variables (e.g. `keyb` and `valueb` in this case)

Reply:
Since config exceptions have not been registered with pybind11, the closest type would be RuntimeError. With __getitem__ and__setitem__ this would change to KeyError